### DISCUSSION TITLE
Replacing KLV in mpegts video

### DISCUSSION THREAD

**Discussion by BigDog6432** [bytedeco/javacpp-presets#1047] [Q&A]
I have an mpegts video and I would like to replace the KLV which appears in the data stream.  Currently when I write it out to a file I am able to look at the contents of the data stream and it appears correct.  However when I stream the video and I look at the packets in wireshark I am not seeing my KLV data.  Just wondering if there are any useful examples that I should be looking at.  

Currently I am creating an AVPacket, initializing it, setting the stream index, setting the data, size and position on the AVPacket.  I them am writing the frame and then passing my AVPacket to av_packet_unref.  

Thanks in advance.

**Comment by saudet** [comment]
FYI, you'll probably get more help about low-level details like that by asking upstream: https://ffmpeg.org/contact.html

**Reply by saudet** [reply]
@bradh Would you have some sample code for this?

**Reply by bradh** [reply]
I don't have a specific example that takes Transport Stream with KLV and produces Transport Stream with KLV, but we have both parts separately. See  https://github.com/WestRidgeSystems/jmisb/tree/main/examples/systemout and https://github.com/WestRidgeSystems/jmisb/tree/main/examples/generator (or more generally, https://github.com/WestRidgeSystems/jmisb).

I didn't write the code, but the specific implementation for turning a metadata blob into AVPacket is at https://github.com/WestRidgeSystems/jmisb/blob/main/api/src/main/java/org/jmisb/api/video/VideoOutput.java#L450-L489

HTH.
